File: src/main/java/com/example/cqrs/query/dto/EventMetadataResponse.java
package com.example.cqrs.query.dto;

import com.example.cqrs.command.entity.event.metadata.EventMetadata;
import lombok.*;

/**
 * 이벤트 메타데이터에 대한 Response DTO입니다.
 */
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class EventMetadataResponse {

    private String correlationId;
    private String causationId;
    private String userId;
    private String eventVersion;

    // factory method
    public static EventMetadataResponse from(EventMetadata metadata) {
        return EventMetadataResponse.builder()
                .correlationId(metadata.getCorrelationId())
                .causationId(metadata.getCausationId())
                .userId(metadata.getUserId())
                .eventVersion(metadata.getEventVersion())
                .build();
    }

}

File: src/main/java/com/example/cqrs/application/account/command/dto/response/EventMetadataResponse.kt
package com.example.cqrs.application.account.command.dto.response

import com.example.cqrs.infrastructure.eventstore.entity.base.metadata.EventMetadata
import com.example.cqrs.infrastructure.eventstore.entity.enumerate.EventSchemaVersion

data class EventMetadataResponse(
    val correlationId: String? = null,
    val causationId: String? = null,
    val userId: String? = null,
    val schemaVersion: EventSchemaVersion
) {
    companion object {
        fun from(eventMetadata: EventMetadata): EventMetadataResponse {
            return EventMetadataResponse(
                correlationId = eventMetadata.correlationId,
                causationId = eventMetadata.causationId,
                userId = eventMetadata.userId,
                schemaVersion = eventMetadata.schemaVersion
            )
        }
    }
}
